main.go

```diff
@@ -15,6 +16,9 @@ var (
 	brokerAddress = kingpin.Flag("mqtt.broker-address", "Address of the MQTT broker.").Default("tcp://localhost:1883").String()
 	topic         = kingpin.Flag("mqtt.topic", "MQTT topic to subscribe to").Default("prometheus/#").String()
 	prefix        = kingpin.Flag("mqtt.prefix", "MQTT topic prefix to remove when creating metrics").Default("prometheus").String()
+	username      = kingpin.Flag("mqtt.username", "MQTT username").Default(os.Getenv("MQTT_USERNAME")).String()
+	password      = kingpin.Flag("mqtt.password", "MQTT password").Default(os.Getenv("MQTT_PASSWORD")).String()
+	clientID      = kingpin.Flag("mqtt.clientid", "MQTT client ID").Default(os.Getenv("MQTT_CLIENT_ID")).String()
```

Comment:
kingping has build in support for environmental parameters, it might not have been turned on yet, but that propably the way to go imho

https://godoc.org/github.com/alecthomas/kingpin#Application.DefaultEnvars

Reply:
That's great! kingpin++, alecthomas++.

I've changed the pull request to use .Envar defaults.